doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+could be done by the syncer. Nevertheless syncers are supposed to contain only
+business logic and should not be impacted by upper layer needs.
+
+#### Definition
+<dl>
+<dt>Parameter adaptation<dt>
+<dd>
+A bijective pure function converting a parameter value between the syncer
+and other parameter reader/writer (including the inference engine).
+
+<why>It must be bijective in order to a) scale the user value to the hardware
+value and b) convert the hardware value to the user's value space.)</why>
+</dd>
+</dl>
+
+#### Requirements
+The following parameter adaptation **SHOULD** be supported
+
+ - Affine adaptation: `affAd(value) = slope * value + offset` where slope and
+   offset and user-defined constants
+   <why>The reference implementation supports it.</why>
+
+ - Logarithm adaptation: `logAd(base, value) = ln(value) / ln(base)` where
+   `ln` is the natural logarithm and base is a user-defined constant.
+   <why>The reference application supports it.</why>
+   <nb>The reference implementation also supports passing a floor value to be
+   applied after conversion.</nb>
+
+A PF **MAY** offer Parameter adaptation composition. I.e. combine multiple parameter
+adaptation
+<nb>E.g.: composing the affine and logarithm adaptation to
+`compAd(value) = slope * logAd(base, value) + offset`.</nb>
+<why>To avoid combination explosion of parameter adaptations. The idea is to
+builtin basic function and let the user compose them to meet its need.</why>
+<ko>The reference application supports in a tricky way: the logarithm
+adaptation is always combined with the affine adaptation</ko>
+
+## Identifiers
+Every parameter **MUST** have an identifier that uniquely identifies it.
+<why>to identify a parameter outside the framework</why>
+
+This identifier **SHOULD** be a string.
+<why>so that the (human</why> user can identify a parameter with ease.)
+
+Two PF instances with the same parameters **MUST** have the same identifier for
+those parameters.
+I.e. this identifier should be the same across all instances with the same
+configuration.
+<why>Persistence of parameter identifier across PF instances with the same
+configuration. To identify parameters independently of the host machine and PF
+instance</why>
+
+### Parameter tree
+A parameter **MUST** be structured in a tree. Each parameter being a distinct
+tree leaf.
+<why>Tree is a simple data structure that can be easily represented and is
+enough to map underlined layers.</why>
+
+Each node of the tree **SHOULD** have its own identifier with the same
+characteristics (type, independence...) than a parameter.
+<why>To represent the tree without treating the leaf nodes specifically.</why>
+
+The identifier of each node of the tree **SHOULD** be a combination of its
+parents. More specifically, if the identifier is a string it should be
+formated the same way as a file system path. E.g. the parameter named
+`/root/child1/4/parameter1`.
```

Comment:
then names cannot contain / ?
seems to be an implentation detail.
does ///.././toto represent a valid path?